toolchain/semantics/semantics_handle.cpp

```diff
@@ -70,10 +71,10 @@ auto SemanticsHandleDesignatedName(SemanticsContext& context,
 
 auto SemanticsHandleDesignatorExpression(SemanticsContext& context,
                                          ParseTree::Node parse_node) -> bool {
-  auto [_, name_id] = context.node_stack().PopForParseNodeAndNameId(
+  auto [_, name_id] = context.node_stack().PopWithParseNode<SemanticsStringId>(
       ParseNodeKind::DesignatedName);
```

Comment:
It seems strange to pop with parse node then discard the parse node. Can you:
```suggestion
  auto name_id = context.node_stack().Pop<SemanticsStringId>(
      ParseNodeKind::DesignatedName);
```

Reply:
Done